### PR TITLE
refactor: treat 404 as an api error, handle 401 and 500 error

### PR DESCRIPTION
**Pull request by ntsmith7**

### CODE REVIEW

File: integration.js [hunk 1]
```diff
@@ -139,23 +180,11 @@ function _lookupEntity(entityObj, options, cb) {
...
-      cb(null, {
-        entity: entityObj,
-        data: null // setting data to null indicates to the server that this entity lookup was a "miss"
-      });
-      return;
+    if (restErr) {
+      return cb(restErr);
     }
 
     if (response.statusCode === 400) {
```
Comment (sarus) [inline integration.js:190]: Should 400 errors be handled as a miss here?  What does confluence return if no results are found.  I'm wondering if we should also handle this 400 error up in the new error handling logic and treat it as an error.  

It's possible that we were getting random 400 errors though and decided to just "swallow" those errors.  Have you seen this happen at all during your development?

**Review by sarus — CHANGES_REQUESTED** [review]
@ntsmith7 Left some comments around the error handling updates.

Thank you!

File: integration.js (hunk 1 above)
Comment (ntsmith7) [reply integration.js:190]: Looking at the documentation,  I don't think the 400s should be handled as a miss - the API will return a 200 for missed results, with an empty array as the body value. I did not run into any random 400 errors while working on this.

The change I'm committing will return an error for 400s.

**Review by sarus — APPROVED** [review]
@ntsmith7 This looks good to me. Thank you for the work on it.  Since this is already in `develop` please move the ticket to QA and we'll have Neil QA off of this `develop` branch.  Once it passes QA then you can merge.

**Pull request merged**